src/OmniSharp.MSBuild/ProjectFile/AnalyzerAssemblyLoader.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Reflection;
+using Microsoft.CodeAnalysis;
+
+namespace OmniSharp.MSBuild.ProjectFile
+{
+    public class AnalyzerAssemblyLoader : IAnalyzerAssemblyLoader
+    {
+        private readonly ConcurrentDictionary<string, Assembly> _assemblyPaths = new ConcurrentDictionary<string, Assembly>();
+
+        public void AddDependencyLocation(string fullPath)
+        {
+            if(!_assemblyPaths.ContainsKey(fullPath))
+                _assemblyPaths.TryAdd(fullPath, Assembly.LoadFrom(fullPath));
```

Comment:
We have an `IAssemblyLoader` abstraction that is probably better to use that `Assembly.LoadFrom(...)` directly here.

Reply:
See https://github.com/OmniSharp/omnisharp-roslyn/blob/master/src/OmniSharp.Abstractions/Services/IAssemblyLoader.cs and https://github.com/OmniSharp/omnisharp-roslyn/blob/master/src/OmniSharp.Host/Services/AssemblyLoader.cs